project/dexter-server/routes/analysis.js

```diff
@@ -1577,10 +1577,13 @@ exports.addV3 = function(req, res) {
         return ;
     }
 
-    var defectJson = req.body.result;
-
     // 1. verify result
-    var defectObject = JSONbig.parse(defectJson);
+    var defectObject;
+    if(typeof req.body.result == "object"){
+        defectObject = req.body.result;
+    } else {
+        defectObject = JSONbig.parse(req.body.result);
```

Comment:
JSONbig.parse can throw an exception for invalid json syntax.
It would be better to add  'try/catch' codes.